astropy/io/fits/file.py

```diff
@@ -578,7 +597,7 @@ def _open_zipfile(self, fileobj, mode):
 
         if close:
             zfile.close()
-        self.compression = 'zip'
+        self._file.seek(0)
```

Comment:
You're right though that there is now a redundant seek in `_open_filelike`, which can be removed.